src/NuGet.Clients/NuGet.CommandLine/GetProjectsReferencingProjectJsonFiles.target

```diff
@@ -3,17 +3,27 @@
             DependsOnTargets="_SplitProjectReferencesByFileExistence"
             Returns="@(_ProjectReferencingProjectJsonFile)">
 
+        <PropertyGroup>
+            <EntryPointProject Condition=" '$(EntryPointProject)' == '' ">$(MsBuildProjectFullPath)</EntryPointProject>
+        </PropertyGroup>
+
         <ItemGroup>
             <_ProjectReferencingProjectJsonFile
+                Include="#:$(MSBuildProjectFullPath)"
+                Condition=" '$(MSBuildProjectFullPath)' == '$(EntryPointProject)' " />
+            <_ProjectReferencingProjectJsonFile
                 Include="$(MSBuildProjectFullPath)"
-                Condition=" Exists('$(MsBuildProjectDirectory)\project.json') " />
+                Condition=" '$(MSBuildProjectFullPath)' != '$(EntryPointProject)' " />
         </ItemGroup>
 
         <MSBuild
           Projects="@(ProjectReference)"
           Targets="_NuGet_GetProjectsReferencingProjectJsonInternal"
           BuildInParallel="$(BuildInParallel)"
-          Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform);"
+          Properties="
+            %(_MSBuildProjectReferenceExistent.SetConfiguration);
+            %(_MSBuildProjectReferenceExistent.SetPlatform);
+            EntryPointProject=$(EntryPointProject)"
```

Comment:
Yup - we do the filtering as part of https://github.com/NuGet/NuGet.Client/pull/107/files#diff-6a9bbd4327c1eb9a97f505940c6a0d8cR279